src/App.tsx

```diff
@@ -1,25 +1,192 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React, { useState, useEffect } from "react";
+import Button from "@mui/material/Button";
+import Typography from "@mui/material/Typography";
+import Grid from "@mui/material/Grid";
+import { Card } from "./component/Card";
+import AppBar from "@mui/material/AppBar";
+
+const CARD_DATA: CardType[] = [
+  { id: 1, design: "♣️" },
+  { id: 2, design: "❤️" },
+  { id: 3, design: "♦️" },
+  { id: 4, design: "♣️" },
+  { id: 5, design: "♠️" },
+  { id: 6, design: "♦️" },
+  { id: 7, design: "♠️" },
+  { id: 8, design: "❤️" },
+  { id: 9, design: "😂" },
+  { id: 10, design: "😂" },
+  { id: 11, design: "🦋" },
+  { id: 12, design: "🦋" },
+  { id: 13, design: "🐶" },
+  { id: 14, design: "🐶" },
+  { id: 15, design: "⭐️" },
+  { id: 16, design: "⭐️" },
+];
+
+export type CardType = {
+  id: number;
+  design: string;
+};
+
+function App(): JSX.Element {
+  const [cards, setCards] = useState<Array<CardType>>([]);
+  const [flippedCards, setFlippedCards] = useState<Array<CardType>>([]);
+  const [matchedCards, setMatchedCards] = useState<Array<CardType>>([]);
+  const [moves, setMoves] = useState<number>(0);
+  const [start, setStart] = useState(true);
```

Comment:
I believe we need flippedCards, matchedCards, start states. We can managed without them.